x-pack/plugins/siem/public/containers/hosts/first_last_seen/index.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import ApolloClient from 'apollo-client';
+import { get } from 'lodash/fp';
+import React, { useEffect, useState } from 'react';
+
+import { GetHostFirstLastSeenQuery } from '../../../graphql/types';
+import { inputsModel } from '../../../store';
+import { QueryTemplateProps } from '../../query_template';
+
+import { HostFirstLastSeenGqlQuery } from './first_last_seen.gql_query';
+
+export interface FirstLastSeenHostArgs {
+  id: string;
+  firstSeen: Date;
+  lastSeen: Date;
+  loading: boolean;
+  refetch: inputsModel.Refetch;
+}
+
+export interface OwnProps extends QueryTemplateProps {
+  children: (args: FirstLastSeenHostArgs) => React.ReactNode;
+  hostName: string;
+}
+
+export function useFirstLastSeenHostQuery<TCache = object>(
```

Comment:
The SIEM App's first custom hook! Thanks for the effort in introducing these and (especially) how to test them to the project @XavierM!  🎉